src/structures/ComponentInteraction.js

```diff
@@ -0,0 +1,61 @@
+'use strict';
+
+const Interaction = require('./Interaction');
+const InteractionResponses = require('./interfaces/InteractionResponses');
+const WebhookClient = require('../client/WebhookClient');
+
+/**
+ * Represents a message button interaction.
+ * @extends {Interaction}
+ */
+class ComponentInteraction extends Interaction {
+  // eslint-disable-next-line no-useless-constructor
+  constructor(client, data) {
+    super(client, data);
+
+    /**
+     * The message to which the button was attached
+     * @type {?Message|Object}
+     */
+    this.message = data.message ? this.channel?.messages.add(data.message) ?? data.message : null;
+
+    /**
+     * The custom ID of the button which was clicked
+     * @type {string}
+     */
+    this.customID = data.data.custom_id;
```

Comment:
There should also be a componentType property if this is to be used for all components. It is always a button for now, and theoretically customIDs would best be unique across all component types so that could be used....but someone is bound to need it.

Reply:
Good point, idk how I missed that one. Will add in.